config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/util/IdValidator.java

```diff
@@ -0,0 +1,136 @@
+package com.ca.apim.gateway.cagatewayconfig.util;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.util.UUID;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class IdValidator {
+    private IdValidator() {
+    }
+
+    static final Pattern PAT_TRAILING_HEX = Pattern.compile("[0-9a-f]+$", Pattern.CASE_INSENSITIVE);
+    static final String[] ZEROS;
+    static final String[] ONES;
+
+    static {
+        StringBuilder sbz = new StringBuilder();
+        StringBuilder sbf = new StringBuilder();
+        String[] z = new String[17];
+        String[] f = new String[17];
+        for (int i = 0; i <= 16; ++i) {
+            z[i] = sbz.toString();
+            sbz.append("0");
+
+            f[i] = sbf.toString();
+            sbf.append("f");
+        }
+
+        ZEROS = z;
+        ONES = f;
+    }
+
+    static String decompressString(@NotNull final String goidString) {
```

Comment:
Honestly speaking, I don't really understand. 
I think GOID means GUID with no hyphens. 
Ex:
e6b0cc9b5613037141b5e1f83679f9ef
a11fb7ba-0c27-4a79-9927-39d3ea3bb3b6
a11fb7ba0c274a79992739d3ea3bb3b6

Reply:
Based on this fact, can't we simplify the logic?

Reply:
This validation is from gateway uneasyrooster